modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/IDPMetadataTestCase.java

```diff
@@ -0,0 +1,281 @@
+            if (federatedAuthenticatorConfigs[i].getName().equals("SAMLSSOAuthenticator")) {
+                samlFederatedAuthenticatorConfig = federatedAuthenticatorConfigs[i];
+                break;
+            }
+        }
+        Assert.assertNotNull(identityProvider);
+        Assert.assertNotNull(samlFederatedAuthenticatorConfig);
+
+        Property [] properties = samlFederatedAuthenticatorConfig.getProperties();
+        for (Property property : properties){
+
+            if(property.getName().equals(IdentityApplicationConstants.Authenticator.SAML2SSO.IDP_ENTITY_ID)){
+                Assert.assertEquals(property.getValue(),ENTITYID);
+            }
+
+            if(property.getName().equals(IdentityApplicationConstants.Authenticator.SAML2SSO.SP_ENTITY_ID)){
+                Assert.assertEquals(property.getValue(), SPENTITYID);
+            }
+
+            if(property.getName().equals(IdentityApplicationConstants.Authenticator.SAML2SSO.SSO_URL)){
+                Assert.assertEquals(property.getValue(), SSOURL);
+            }
+
+            if(property.getName().equals(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_AUTHN_REQ_SIGNED)){
+                Assert.assertEquals(property.getValue(), IS_AUTHN_REQ_SIGNED);
+            }
+
+            if(property.getName().equals(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_LOGOUT_ENABLED)){
+                Assert.assertEquals(property.getValue(), IS_LOGOUT_ENABLED);
+            }
+
+            if(property.getName().equals(IdentityApplicationConstants.Authenticator.SAML2SSO.LOGOUT_REQ_URL)){
+                Assert.assertEquals(property.getValue(), LOGOUT_REQ_URL);
+            }
+
+            if(property.getName().equals(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_ENABLE_ASSERTION_ENCRYPTION)){
+                Assert.assertEquals(property.getValue(), IS_ENABLE_ASSERTION_ENCRYPTION);
+            }
+
+            if(property.getName().equals(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_ENABLE_ASSERTION_SIGNING)){
+                Assert.assertEquals(property.getValue(), IS_ENABLE_ASSERTION_SIGNING);
+            }
+        }
+
+    }
+
+    @AfterClass
+    public void endTest() throws RemoteException, IdentitySAMLSSOConfigServiceIdentityException {
+        try {
+            identityProviderMgtServiceClient.deleteIdP(ENTITYID);
+        } catch (Exception ex) {
+            throw new RemoteException("Error Deleting Identity Provider", ex);
+        }
+    }
+
+
+    private static void buildSAMLAuthenticationConfiguration(IdentityProvider fedIdp, String metadataXML) {
+
+        FederatedAuthenticatorConfig saml2SSOAuthnConfig = new FederatedAuthenticatorConfig();
+        saml2SSOAuthnConfig.setName("SAMLSSOAuthenticator");
+        saml2SSOAuthnConfig.setDisplayName("samlsso");
+        saml2SSOAuthnConfig.setEnabled(true);
+        Property[] properties = new Property[25];
+
+        Property property = new Property();
```

Comment:
why do we need all these configs? If we upload a metadata file aren't we configuring the "Service Provider Entity Id" only?